source/extensions/filters/http/rbac/rbac_filter.cc

```diff
@@ -122,8 +115,9 @@ RoleBasedAccessControlFilter::decodeHeaders(Http::RequestHeaderMap& headers, boo
     } else {
       ENVOY_LOG(debug, "enforced denied, matched policy {}",
                 effective_policy_id.empty() ? "none" : effective_policy_id);
+      callbacks_->streamInfo().setResponseFlag(StreamInfo::ResponseFlag::UnauthorizedRBAC);
       callbacks_->sendLocalReply(Http::Code::Forbidden, "RBAC: access denied", nullptr,
-                                 absl::nullopt, RcDetails::get().RbacAccessDenied);
+                                 absl::nullopt, effective_policy_id);
```

Comment:
I think this might lead to dangling string_view. Can you check that policy id is used before it goes out of scope?

Reply:
It's used in several places:

1. sendLocalReply() -> setResponseCodeDetails() and then it calls `response_code_details_.emplace(rc_details);` where `response_code_details_` is defined as `absl::optional<std::string>`  (https://github.com/envoyproxy/envoy/blob/master/source/common/http/filter_manager.cc#L324-L330)

2. sendLocalReplyViaFilterChain() and then it just `ENVOY_STREAM_LOG(debug, "Sending local reply with details {}", *this, details);` (https://github.com/envoyproxy/envoy/blob/master/source/common/http/filter_manager.cc#L773)

---

it looks like the string_view is not used after returning from sendLocalReply(), but it also looks like we're using this `ConstSingleton<RcDetailsValues>` everywhere which seems indicating the string_view could be used after the scope? @alyssawilk do you have suggestions here? thanks

Reply:
the two things which are done is a logged error (inline in the stack, fine) and setting the response code details in the stream info for later logging (optional string, does its own copy) so I think this is fine.